tools/api.py

```diff
@@ -1057,9 +1062,11 @@ def add_cpp_class(self) -> None:
     ${output_decls}
     if (${fname} == nullptr)
         throw std::runtime_error("${name} function is missing.");
+    const char* exception_msg = " ";
     auto api_error_result = ${fname}(${args});
+    const std::string exception_str(exception_msg); 
```

Comment:
We should only construct the string when there is an error. So this could be moved to the if branch below.